File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.py
import sys
input = sys.stdin.readline
formula = input().rstrip()
array = []
temp = ""
for str in formula:
    if str == "-" or str == "+":
        array.append(temp)
        array.append(str)
        temp = ""
        continue
    temp += str
array.append(temp)

result = 0
temp = 0
flag = -1
for index, element in enumerate(array):
    if element == '-':
        flag = 1
        result -= temp
        temp = 0
        continue
    elif element == '+':
        continue
    if flag == 1:
        temp += int(element)
        if index == len(array)-1:
            result -= temp
    else:
        result += int(element)
print(result)

File: 백준/Silver/1541. 잃어버린 괄호/잃어버린 괄호.swift
let formula = readLine()!
var array = [String]()
var newArray = [String]()
var temp = ""
var flag = false
var result = 0

for char in formula {
    if char == "-" || char == "+" {
        array.append(temp)
        array.append(String(char))
        temp = ""
    } else {
        temp += String(char)
    }
}
array.append(temp)

for element in array {
    if element == "+" {
        newArray.removeLast()
        flag = true
    } else if flag == true  {
        let result = String(Int(temp)! + Int(element)!)
        newArray.append(result)
        temp = result
        flag = false
    } else {
        temp = element
        newArray.append(element)
    }
}

for i in 0..<newArray.count {
    if i == 0 {
        result = Int(newArray[i])!
        continue
    }

    if let number = Int(newArray[i]) {
        result -= number
    }
}

print(result)